File: ws/lascano/u2/ws17Libraries/StoreApp/src/ec/edu/espe/store/model/Product.java
package ec.edu.espe.store.model;

import utils.Tax;

/**
 *
 * @author Edison Lascano, Object Masters, @ESPE
 */
public class Product {

    private int id;
    private String description;
    private float price;
    private float pvp;

    public Product(int id, String description, float price) {
        this.id = id;
        this.description = description;
        this.price = price;

        //TODO compute total price
        pvp = Tax.computeTotal(price, 15F);
    }

    public Product(int id, String description, float price, float pvp) {
        this.id = id;
        this.description = description;
        this.price = price;
        this.pvp = pvp;
    }

    @Override
    public String toString() {
        return "Product\n{" + "id=" + id + "\n description=" + description + "\n price=" + price + "\n pvp=" + pvp + '}';
    }

    /**
     * @return the id
     */
    public int getId() {
        return id;
    }

    /**
     * @param id the id to set
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * @return the description
     */
    public String getDescription() {
        return description;
    }

    /**
     * @param description the description to set
     */
    public void setDescription(String description) {
        this.description = description;
    }

    /**
     * @return the price
     */
    public float getPrice() {
        return price;
    }

    /**
     * @param price the price to set
     */
    public void setPrice(float price) {
        this.price = price;
    }

    /**
     * @return the pvp
     */
    public float getPvp() {
        return pvp;
    }

    /**
     * @param pvp the pvp to set
     */
    public void setPvp(float pvp) {
        this.pvp = pvp;
    }

}

File: my-fresh-copy/hw/tipantiza/u2/hw09Libraries/StoreApp_PY/model/product.py
# model/product.py

from utils.tax import Tax

class Product:

    def __init__(self, id, description, price, pvp=None):
        self.id = id
        self.description = description
        self.price = price

        
        if pvp is None:
            self.pvp = Tax.compute_total(price, 15)
        else:
            self.pvp = pvp

    def __str__(self):
        return (
            "Product{\n"
            f"  id={self.id}\n"
            f"  description={self.description}\n"
            f"  price={self.price}\n"
            f"  pvp={self.pvp}\n"
            "}"
        )